src/Aspire.Hosting.AWS.Provisioning/CloudFormation/CloudFormationProvisionerExtensions.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.AWS.Provisioning.CloudFormation;
+using Aspire.Hosting.Lifecycle;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Aspire.Hosting;
+
+public static class CloudFormationProvisionerExtensions
+{
+    /// <summary>
+    /// Add a CloudFormation stack for provisioning application resources.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="stackName">The name of the CloudFormation stack.</param>
+    /// <param name="templatePath">The path to the CloudFormation template that defines the CloudFormation stack.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<ICloudFormationResource> AddAWSCloudFormationProvisioning(this IDistributedApplicationBuilder builder, string stackName, string templatePath)
+    {
+        var resource = new CloudFormationResource(stackName, templatePath);
+        var cfBuilder = builder.AddResource(resource);
+
+        builder.Services.AddLifecycleHook<CloudFormationLifeCycle>(sp => ActivatorUtilities.CreateInstance<CloudFormationLifeCycle>(sp, resource));
```

Comment:
Any reason this doesn't follow the same pattern as the other code? Singleton that looks at all resources instead of passing the resource here?